File: backend/app/Models/SesionEstudio.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class SesionEstudio extends Model
{
    protected $fillable = [
        'titulo',
        'lugar',
        'fecha_hora',
        'descripcion',
        'materia_id',
        'user_id',
    ];

    public function materia()
    {
        return $this->belongsTo(Materia::class);
    }

    public function creador()
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    public function participantes()
    {
        return $this->belongsToMany(User::class, 'sesion_participantes');
    }
}

File: android/app/src/main/java/com/ucsc/conectaucsc/data/model/SesionEstudio.kt
package com.ucsc.conectaucsc.data.model

import kotlinx.serialization.Serializable

@Serializable
data class SesionEstudio(
    val id: Int,
    val titulo: String,
    val lugar: String,
    val fecha_hora: String,
    val descripcion: String?,
    val materia_id: Int,
    val user_id: Int,
    val creador: UsuarioSimple?,
    val participantes: List<UsuarioSimple>?,
    val materia: MateriaSimple? = null
)

@Serializable
data class UsuarioSimple(
    val id: Int,
    val name: String
)

data class MateriaSimple(
    val id: Int,
    val nombre: String
)

@Serializable
data class CrearSesionRequest(
    val titulo: String,
    val lugar: String,
    val fecha_hora: String,
    val descripcion: String?,
    val materia_id: Int
)